code/datums/components/forensics.dm

```diff
@@ -44,7 +44,7 @@
 /datum/component/forensics/proc/wipe_blood_DNA()
 	blood_DNA = null
 	if(isitem(parent))
-		qdel(parent.GetComponent(/datum/component/decal/blood))
+		SEND_SIGNAL(src, COMSIG_WIPE_BLOOD_DNA)
```

Comment:
The signal should be generic, not dependent on whether the parent is an item or not.
Can you add the if check on whether to register to the signal in the first place instead?